pkg/telemetry/phonehome/request_params.go

```diff
@@ -0,0 +1,59 @@
+package phonehome
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/stackrox/rox/pkg/grpc/authn"
+)
+
+// RequestParams holds intercepted call parameters.
+type RequestParams struct {
+	UserAgent string
+	UserID    authn.Identity
+	Path      string
+	Code      int
+	GRPCReq   any
+	HTTPReq   *http.Request
+}
+
+// GetProtocol returns HTTP for requests with HTTP request and gRPC otherwise.
+func (rp *RequestParams) GetProtocol() string {
```

Comment:
I think this is more of an edge case and not something we necessarily need to address now:

We provide the option to also use gRPC when the server is exposed behind load balancers which do not necessarily support it (there's https://github.com/stackrox/go-grpc-http1 which covers this).

This means, internally while the API call _may_ be a gRPC one, the protocol it actually uses _may_ be something different (e.g. wss).

Internally, we use this for all sorts of clients (e.g. roxctl, sensor <-> central, etc.). I don't know the full history about this, but probably we ran into some issues within some environments, which lead to us coming up with this solution.

I'm not sure whether there's the desire to also collect the _specific_ protocol used, to further determine the sort of environments customers are using, but it might be worth to consider in the future (essentially, what I'm trying to say is that protocol `gRPC` a.k.a. `http2` _may not be correct_ for all gRPC calls).

Reply:
I agree that the existing protocol detection is misleading. And, as was not explicitly requested, I'd rather remove this field from the report.